File: src/main/java/logic/PlayerStatus.java
package logic;

import lib.ConfigurableOption;
import lib.DrawingUtility;
import lib.GameManager;
import lib.IRenderableObject;

public class PlayerStatus /* fill code */ {

	private int remainingTime = 0;
	private int score = 0;
	private Gun currentGun = null;
	private boolean pause = false;
	
	public PlayerStatus(){
		/* fill code */
	}

	/* fill required getter & setter */

	public void decreaseRemainingTime(int amount) {
		/* fill code */ 
	}
	
	public void increaseScore(int score) {
		/* fill code */;
	}

	public boolean isDisplayingArea(int x,int y){
		return y < 40;
	}

	/* fill code interface methods */
	
}

File: src/logic/PlayerStatus.kt
package logic

import lib.ConfigurableOption
import lib.DrawingUtility
import lib.GameManager
import lib.IRenderableObject

class PlayerStatus() : IRenderableObject {
    var remainingTime = ConfigurableOption.timelimit * GameManager.TICK_PER_SECONDS
        private set
    var score = 0
        private set
    lateinit var currentGun: Gun
    var pause = false

    fun decreaseRemainingTime(amount: Int) {
        remainingTime -= amount
    }

    fun increaseScore(score: Int) {
        this.score += score
    }

    fun isDisplayingArea(x: Int, y: Int) = y < 40

    override fun isVisible() = true

    override fun getZ() = Int.MAX_VALUE

    override fun render() = DrawingUtility.drawStatusBar(remainingTime / GameManager.TICK_PER_SECONDS, score, currentGun, pause)
}
